keyboards/dumbpad/v3x/v3x.c

```diff
@@ -0,0 +1,125 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "v3x.h"
+
+#ifdef RGB_MATRIX_ENABLE
+led_config_t g_led_config = {{// Key Matrix to LED Index
+                              {NO_LED, 0, 1, 2, 3},
+                              {NO_LED, 7, 6, 5, 4},
+                              {NO_LED, 8, 9, 10, 11},
+                              {NO_LED, 15, 14, 13, 12}},
+                             {// LED Index to Physical Position
+                              {0, 0},
+                              {75, 0},
+                              {149, 0},
+                              {224, 0},
+
+                              {224, 21},
+                              {149, 21},
+                              {75, 21},
+                              {0, 21},
+
+                              {0, 43},
+                              {75, 43},
+                              {149, 43},
+                              {224, 43},
+
+                              {224, 64},
+                              {149, 64},
+                              {75, 64},
+                              {0, 64}},
+                             {// LED Index to Flag
+                              4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4}};
+#endif
+
+void keyboard_pre_init_kb(void) {
+    // Set LED IO as outputs
+    setPinOutput(LED_00);
+    setPinOutput(LED_01);
+    setPinOutput(LED_02);
+    keyboard_pre_init_user();
+}
+
+void shutdown_user() {
+    // Shutdown LEDs
+    writePinLow(LED_00);
+    writePinLow(LED_01);
+    writePinLow(LED_02);
+}
+
+layer_state_t layer_state_set_kb(layer_state_t state) {
+    // Layer LEDs act as binary indication of current layer
+    uint8_t layer = get_highest_layer(state);
+    writePin(LED_00, layer & 0b1);
+    writePin(LED_01, (layer >> 1) & 0b1);
+    uprintf("%d string", layer);
+    return layer_state_set_user(state);
+}
+
+// Optional override functions below.
+// You can leave any or all of these undefined.
+// These are only required if you want to perform custom actions.
+
+void matrix_init_kb(void) {
+    // put your keyboard start-up code here
+    // runs once when the firmware starts up
+    uint8_t led_delay_ms = 80;
+    for (int i = 0; i < 2; i++) {
+        writePinHigh(LED_00);
+        writePinHigh(LED_01);
+        writePinHigh(LED_02);
+        wait_ms(led_delay_ms);
+        writePinLow(LED_00);
+        writePinLow(LED_01);
+        writePinLow(LED_02);
+        if (i < 1) {
+            wait_ms(led_delay_ms);
+        }
+    }
+
+    matrix_init_user();
+}
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
```

Comment:
```suggestion
#ifdef ENCODER_ENABLE
bool encoder_update_user(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
```

Reply:
@drashna  / @zvecr sorry, I mixed up the `encoder_update_user` and `encoder_update_kb` implementations, should be ok now :)

* `encoder_update_kb` => in keyboard.c, call `encoder_update_user`
* `encoder_update_user` => in keymaps

correct (similar to planck).